galley/pkg/config/analysis/analyzers/analyzers_test.go

```diff
@@ -138,6 +138,15 @@ var testGrid = []testCase{
 			{msg.IstioProxyVersionMismatch, "Pod/enabled-namespace/details-v1-pod-old"},
 		},
 	},
+	{
+		name:       "virtualServiceConflictingHosts",
+		inputFiles: []string{"testdata/virtualservice_conflictinghosts.yaml"},
+		analyzer:   &virtualservice.ConflictingHostsAnalyzer{},
+		expected: []message{
+			{msg.ConflictingVirtualServiceHosts, ""},
+			{msg.ConflictingVirtualServiceHosts, ""},
```

Comment:
@sushicw the current way of verifying messages in unit tests is insufficient for this analyzer. We should verify the entire string message in my opinion.

Reply:
Maybe, I think having string comparison in tests forces us to follow good practices and make sure that the template string after substitution still makes sense for users.

I can add the optional third string.